components/CustomSelect/index.tsx

```diff
@@ -48,31 +57,38 @@ const CustomSelect = <T extends string | number | boolean>({
     };
   }, []);
 
+  const componentId = id ?? `${label}-CustomSelect`;
+
   return (
-    <SelectContainer
-      ref={containerRef}
-      onClick={isContainerClickable ? () => setIsOpen(!isOpen) : undefined}
-    >
-      <SelectedValue>
-        {options.find(option => option.value === value)?.label || label}
-      </SelectedValue>
-      <DropdownIcon onClick={() => setIsOpen(!isOpen)}>
-        <Icon type="pencil" />
-      </DropdownIcon>
-      {isOpen && (
-        <OptionsContainer>
-          {options.map(option => (
-            <Option
-              key={String(option.value)}
-              onClick={() => handleOptionClick(option.value)}
-            >
-              {option.label}
-            </Option>
-          ))}
-        </OptionsContainer>
+    <Flex $direction="column" $gap="4px">
+      {label && (
+        <P2 as="label" htmlFor={componentId}>
+          {label}
+        </P2>
       )}
-    </SelectContainer>
+      <SelectContainer
+        ref={containerRef}
+        onClick={isContainerClickable ? () => setIsOpen(!isOpen) : undefined}
+      >
+        <P2 $color={COLORS.midgray}>
+          {options.find(option => option.value === value)?.label || placeholder}
```

Comment:
we may want to refactor `DropdownOption<T>` to `Record<T, string>`, i.e. use the values as keys, to mitigate needing to search through `options` for the corresponding label.